### DISCUSSION TITLE
Sesion Storage Space (S3)

### DISCUSSION THREAD

**Discussion by graficar** [devlikeapro/waha#259] [Q&A]
Hello,

How can I store sessions in Digital Ocean Space?
I have the plus version.

Thanks!

[![patron:ADVANCED](https://img.shields.io/badge/patron-ADVANCED-4cd07d)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
@graficar Hi! 

You need to mount the Digital Ocean Space (or any S3 compatible storage - AWS S3 or self-hosted minio) as linux filesystem in order to make it work.

https://bluexp.netapp.com/blog/amazon-s3-as-a-file-system
If you mount a bucket to `/home/username/s3-sessions/` folder, now you can run a docker image and tell it to save sessions (authorization) data there https://waha.devlike.pro/docs/how-to/sessions/#session-persistent

```bash
docker run -d -v /home/username/s3-sessions/:/app/.sessions -p 3000:3000/tcp --name whatsapp-http-api devlikeapro/whatsapp-http-api-plus
```

Remember about the notes tho https://bluexp.netapp.com/blog/amazon-s3-as-a-file-system
> ## Considerations When Using Amazon S3 as a File System
> - When you are using Amazon S3 as a file system, you might observe a network delay when performing IO centric operations such as creating or moving new folders or files. The performance depends on your network speed as well distance from Amazon S3 storage region.
> - Since Amazon S3 is not designed for atomic operations, files cannot be modified, they have to be completely replaced with modified files. This doesn’t impact your application as long as it’s creating or deleting files; however, if there are frequent modifications to a file, that means replacing the file on Amazon S3 repeatedly, which results in multiple put requests and, ultimately, higher costs.
> - As files are transferred via HTTPS, whenever your application tries to access the mounted Amazon S3 bucket first time, there is noticeable delay. Future or subsequent access times can be delayed with local caching.

----

The number one task for the project in this year is adding [remote storage support](#41) but likely it'll work only with mongodb (because we also want to use it for storing messages and other events there)

I hope it answers your questions!

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by graficar** [reply]
The problem I have is that I am not mounting a droplet with a Linux system, but rather I am using the APPS service that allows mounting a Docker Hub image without having to manage the server. Therefore I don't run the docker run command manually and I don't have persistent disk space either.

The other alternative that I would have is to store the sessions in a database. How can I do that?

[![patron:ADVANCED](https://img.shields.io/badge/patron-ADVANCED-4cd07d)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by allburov** [comment]
> is to store the sessions in a database

Sorry, we put the wrong link above. We're working on that, hopefully in 1-2 months it'll be ready!
https://github.com/devlikeapro/whatsapp-http-api/issues/41

**Reply by graficar** [reply]
ok thanks

[![patron:ADVANCED](https://img.shields.io/badge/patron-ADVANCED-4cd07d)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
